index.js

```diff
@@ -1,155 +1,316 @@
-const { existsSync, readFileSync } = require('fs');
-const { join } = require('path');
+/* tslint:disable */
+/* eslint-disable */
+/* prettier-ignore */
 
-const { platform, arch } = process;
+/* auto-generated by NAPI-RS */
 
-let nativeBinding = null;
-let localFileExisted = false;
-let loadError = null;
+const { existsSync, readFileSync } = require('fs')
+const { join } = require('path')
+
+const { platform, arch } = process
+
+let nativeBinding = null
+let localFileExisted = false
+let loadError = null
 
 function isMusl() {
   // For Node 10
   if (!process.report || typeof process.report.getReport !== 'function') {
     try {
-      return readFileSync('/usr/bin/ldd', 'utf8').includes('musl');
+      const lddPath = require('child_process').execSync('which ldd').toString().trim()
+      return readFileSync(lddPath, 'utf8').includes('musl')
     } catch (e) {
-      return true;
+      return true
     }
   } else {
-    const { glibcVersionRuntime } = process.report.getReport().header;
-    return !glibcVersionRuntime;
+    const { glibcVersionRuntime } = process.report.getReport().header
+    return !glibcVersionRuntime
   }
 }
 
 switch (platform) {
+  case 'android':
+    switch (arch) {
+      case 'arm64':
+        localFileExisted = existsSync(join(__dirname, 'zstd.android-arm64.node'))
+        try {
+          if (localFileExisted) {
+            nativeBinding = require('./zstd.android-arm64.node')
+          } else {
+            nativeBinding = require('@mongodb-js/zstd-android-arm64')
+          }
+        } catch (e) {
+          loadError = e
+        }
+        break
+      case 'arm':
+        localFileExisted = existsSync(join(__dirname, 'zstd.android-arm-eabi.node'))
+        try {
+          if (localFileExisted) {
+            nativeBinding = require('./zstd.android-arm-eabi.node')
+          } else {
+            nativeBinding = require('@mongodb-js/zstd-android-arm-eabi')
+          }
+        } catch (e) {
+          loadError = e
+        }
+        break
+      default:
+        throw new Error(`Unsupported architecture on Android ${arch}`)
+    }
+    break
   case 'win32':
     switch (arch) {
       case 'x64':
-        localFileExisted = existsSync(join(__dirname, 'zstd.win32-x64-msvc.node'));
+        localFileExisted = existsSync(
+          join(__dirname, 'zstd.win32-x64-msvc.node')
+        )
         try {
           if (localFileExisted) {
-            nativeBinding = require('./zstd.win32-x64-msvc.node');
+            nativeBinding = require('./zstd.win32-x64-msvc.node')
           } else {
-            nativeBinding = require('@mongodb-js/zstd-win32-x64-msvc');
+            nativeBinding = require('@mongodb-js/zstd-win32-x64-msvc')
           }
         } catch (e) {
-          loadError = e;
+          loadError = e
```

Comment:
How come we are removing semicolons at the ends of a lot of these lines?

Reply:
(see the other thread)